front/context/src/ApiContext.tsx

```diff
@@ -30,25 +27,32 @@ export function ApiContextProvider(
   props: ApiContextProviderProps
 ): React.ReactElement {
   const { children = null, provider } = props;
-  const [state, setState] = useState<State>({ isApiReady: false });
-  const { isApiReady } = state;
+  const [api, setApi] = useState<ApiRx>(new ApiRx({ provider }));
+  const [isReady, setIsReady] = useState(false);
 
-  const apiRef = useRef(new ApiRx({ provider }));
-  const api = apiRef.current;
+  useEffect(() => {
+    // We want to fetch all the information again each time we reconnect. We
+    // might be connecting to a different node, or the node might have changed
+    // settings.
+    setApi(new ApiRx({ provider }));
+    setIsReady(false);
+  }, [provider]);
 
   useEffect(() => {
     // We want to fetch all the information again each time we reconnect. We
     // might be connecting to a different node, or the node might have changed
     // settings.
-    api.isReady.subscribe(() => {
+    const subscription = api.isReady.subscribe(() => {
       l.log(`Api ready, app is ready to use`);
 
-      setState({ isApiReady: true });
+      setIsReady(true);
     });
-  }, [api, provider]);
+
+    return () => subscription.unsubscribe();
+  });
```

Comment:
```suggestion
  }, [api]);
```
^ i would think this to be necessary